dojo/tools/hcl_asoc_sast/parser.py

```diff
@@ -0,0 +1,151 @@
+from xml.dom import NamespaceErr
+
+from defusedxml import ElementTree as ET
+
+from dojo.models import Finding
+
+
+class HCLASoCSASTParser:
+    def get_scan_types(self):
+        return ["HCL AppScan on Cloud SAST XML"]
+
+    def get_label_for_scan_types(self, scan_type):
+        return scan_type
+
+    def get_description_for_scan_types(self, scan_type):
+        return "Import XML output of HCL AppScan on Cloud SAST"
+
+    def xmltreehelper(self, input):
+        if input.text is None:
+            output = None
+        elif "\n" in input.text:
+            output = ""
+            for i in input:
+                output = output + " " + i.text
+        else:
+            output = " " + input.text
+        return output
+
+    def get_findings(self, file, test):
+        findings = []
+        tree = ET.parse(file)
+        root = tree.getroot()
+        if "xml-report" not in root.tag:
+            msg = "This doesn't seem to be a valid HCL ASoC SAST xml file."
+            raise NamespaceErr(msg)
+        report = root.find("issue-group")
+        if report is not None:
+            for finding in report:
+                title = ""
+                description = ""
+                for item in finding:
+                    match item.tag:
+                        case "severity":
+                            output = self.xmltreehelper(item)
+                            severity = "Info" if output is None else output.strip(" ").capitalize()
+                        case "cwe":
+                            cwe = int(self.xmltreehelper(item))
+                        case "issue-type":
+                            title = self.xmltreehelper(item).strip()
+                            description = description + "***Issue-Type:" + title + "\n"
```

Comment:
The Markdown here isn't closing the bold tags on the same line, resulting in lines alternating back and forth in their formatting as subsequent lines close the prior line's tags rather than opening their own. I've provided suggestions for some additional examples below, but they'll all need to be updated in this way.

![Screenshot 2024-12-19 at 11 32 26](https://github.com/user-attachments/assets/124a8763-5197-4662-80ab-e1e471151b82)


```suggestion
                            description = description + "**Issue-Type:** " + title + "\n"
```